src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -31,13 +31,11 @@ def __init__(self, argv):
 
     def get_command_table(self):
         import azure.cli.core.commands as commands
-        # Find the first noun on the command line and only load commands from that
```

Comment:
This optimization also put a limitation on how commands are loaded.
It meant that commands from other modules could not contribute to a top-level command because of this shortcut.

If the overhead of loading all modules is not high, it makes sense to do this and that's the aim of this change.